File: LISTA 1/Exercício 4/Exercicio 4 (RPC)/Servidor04.py
from xmlrpc.server import SimpleXMLRPCServer
class RPC:
    _metodos_rpc = ['classificacao']
    def __init__(self, direcao):
        self._servidor = SimpleXMLRPCServer(direcao, allow_none=True)
        for metodo in self._metodos_rpc:
            self._servidor.register_function(getattr(self, metodo))

    def classificacao(self, altura, sexo):
        print(sexo)
        peso_ideal = 0
        if sexo == 'masculino':   
            peso_ideal = (72.7*float(altura))-58
            return "O peso ideal para ele é ", peso_ideal
        elif sexo == 'feminino':
            peso_ideal = (62.1*float(altura))-44.7
            return "O peso ideal para ela é ", peso_ideal
        else:
            return "Informação errada, tente novamente"
    def iniciar_servidor(self):
        self._servidor.serve_forever()
if __name__ == '__main__' :
    rpc = RPC(('', 20064))
    print("Servidor RPC iniciado ... ")
    rpc.iniciar_servidor()

File: LISTA 1/Exercício 4/Exercicio 4 (Sockets)/Servidor04.java
import java.io.DataInputStream;
import java.io.DataOutputStream;
import java.io.IOException;
import java.io.OutputStream;
import java.io.InputStream;
import java.net.ServerSocket;
import java.net.Socket;

public class Servidor04 {
	public static void main(String[] args) throws IOException {
		ServerSocket ss = new ServerSocket(7777);
		System.out.println("Servidor esperando por conexoes...");
		Socket socket = ss.accept(); 
		System.out.println("Conexao de " + socket + "!");

		InputStream inputStream = socket.getInputStream();
		DataInputStream dataInputStream = new DataInputStream(inputStream);
		
		OutputStream outputStream = socket.getOutputStream();
		DataOutputStream dataOutputStream = new DataOutputStream(outputStream);

		double altura = dataInputStream.readDouble();
		double sexo = dataInputStream.readDouble();

		dataOutputStream.writeUTF(classificacao(altura, sexo));
		dataOutputStream.flush();
		
		dataOutputStream.close();
		
		System.out.println("Fechando Sockets.");
		ss.close();
		socket.close();
	}
	
	public static String classificacao(double altura, double sexo){
		String resposta;
		Double peso_ideal;
		if(sexo==1){
			peso_ideal = (72.7*altura)-58;
			resposta = "O peso ideal para ele é " + peso_ideal; 
		}
		else if(sexo==2){
			peso_ideal = (62.1*altura)-44.7;
			resposta = "O peso ideal para ela é " + peso_ideal;
		} 
		else 
			resposta = "Informação errada, tente novamente";
		return resposta;
	}
}